client/src/pages/course/score.tsx

```diff
@@ -44,8 +54,32 @@ export function Page(props: CoursePageProps) {
     }
   }, []);
 
+  const setQueryParams = useCallback(
+    (query: ParsedUrlQuery) => {
+      router.push(
+        {
+          pathname: router.pathname,
+          query,
+        },
+        undefined,
+        {
+          shallow: true,
+        },
+      );
+    },
+    [router],
+  );
+
   const getCourseScore = useCallback(
     async (pagination: IPaginationInfo, filters: ScoreTableFilters, order: ScoreOrder) => {
+      const { cityName } = filters;
+      if (isNull(cityName)) {
+        const newQuery = { ...router.query };
+        delete newQuery.cityName;
+        setQueryParams(newQuery);
```

Comment:
Please consider simplify it
```suggestion
        const { cityName, ...newQuery } = router.query;
        setQueryParams(newQuery);
```